addons/ui/org.openhab.ui.cometvisu/src/main/java/org/openhab/ui/cometvisu/internal/util/ClientInstaller.java

```diff
@@ -0,0 +1,306 @@
+                    latestRelease = jsonResponse.get(0);
+                }
+            } catch (IOException e) {
+                logger.error("error downloading release data: {}", e.getMessage(), e);
+            }
+        }
+        return latestRelease;
+    }
+
+    /**
+     * Download the latest release and extract it to the configured COMETVISU_WEBFOLDER
+     */
+    public void downloadLatestRelease() {
+        // request the download URL for the latest CometVisu release from the github API
+        Map latestRelease = getLatestRelease();
+        ArrayList assets = (ArrayList) latestRelease.get("assets");
+        String downloadUrl = "";
+        Map releaseAsset = null;
+        for (Object assetObj : assets) {
+            Map asset = (Map) assetObj;
+            if (((String) asset.get("content_type")).equalsIgnoreCase("application/zip")) {
+                releaseAsset = asset;
+                break;
+            }
+        }
+        if (releaseAsset == null) {
+            logger.error("no zip download file found for release {}", latestRelease.get("name"));
+        } else {
+            File releaseFile = new File("release.zip");
+            try {
+                URL url = new URL((String) releaseAsset.get("browser_download_url"));
+
+                FileUtils.copyURLToFile(url, releaseFile);
+
+                ZipFile zip = new ZipFile(releaseFile, ZipFile.OPEN_READ);
+
+                extractFolder("cometvisu/release/", zip, Config.COMETVISU_WEBFOLDER);
+            } catch (IOException e) {
+                logger.error("error opening release zip file {}", e.getMessage(), e);
+            } finally {
+                if (releaseFile.exists()) {
+                    releaseFile.delete();
+                }
+            }
+        }
+    }
+
+    /**
+     * Extract the content of the zip file to the target folder
+     *
+     * @param folderName {String} subfolder inside the zip file that should be extracted
+     * @param zipFile {ZipFile} zip-file to extract
+     * @param destDir {String} destination for the extracted files
+     */
+    private void extractFolder(String folderName, ZipFile zipFile, String destDir) {
+        for (ZipEntry entry : Collections.list(zipFile.entries())) {
+            if (entry.getName().startsWith(folderName)) {
+                String target = entry.getName().substring(folderName.length());
+                File file = new File(destDir, target);
+                if (entry.isDirectory()) {
+                    file.mkdirs();
+                } else {
+                    if (file.exists() && file.getPath().matches(".*/config/visu_config.*\\.xml")) {
+                        // never ever overwrite existing config files
+                        continue;
+                    }
+                    new File(file.getParent()).mkdirs();
+
+                    InputStream is = null;
+                    OutputStream os = null;
+
+                    try {
```

Comment:
Consider using try-with-resources this could simplify the closing of the connections